File: Python/String-1/middle_three.py

# given a string of odd length, return the string length 3 from its middle
# the string length will be at least 3
def middle_three(str):
    s = ''
    middle = (int)(len(str)/2)
    
    s = str[middle-1:middle+2]

    return s

print(middle_three('Candy'))
print(middle_three('and'))
print(middle_three('solving'))

File: JavaScript/String-1/middleThree.js

// given a string of odd length, return the string length 3 from its middle, so 'Candy' yields 'and'
// the string length will be at least 3
const middleThree = (str) => {
    return str.substring((str.length / 2) - 1, (str.length / 2) + 2);
};

console.log(middleThree('Candy'));
console.log(middleThree('and'));
console.log(middleThree('solving'));
